airbyte-ci/connectors/pipelines/pipelines/helpers/connectors/modifed.py

```diff
@@ -21,24 +21,37 @@ def get_connector_modified_files(connector: Connector, all_modified_files: Set[P
     return frozenset(connector_modified_files)
 
 
-def _find_modified_connectors(
-    file_path: Union[str, Path], active_connectors: Set[Connector], dependency_scanning: bool = True
-) -> Set[Connector]:
-    """Find all connectors impacted by the file change."""
-    modified_connectors = set()
+def _is_connector_modified_directly(connector: Connector, modified_files: Set[Union[str, Path]]) -> bool:
+    """Test if the connector is being impacted by file changes in the connector itself."""
+    result = False
+    for file_path in modified_files:
+        if _is_ignored_file(file_path):
+            continue
 
-    for connector in active_connectors:
         if Path(file_path).is_relative_to(Path(connector.code_directory)) or file_path == connector.documentation_file_path:
             main_logger.info(f"Adding connector '{connector}' due to connector file modification: {file_path}.")
-            modified_connectors.add(connector)
+            result = True
```

Comment:
```suggestion
            return True
```

Reply:
You could exit the for loop early to spare iteration on other files if there's a match

Reply:
done